exporter/dynatraceexporter/metrics_exporter.go

```diff
@@ -94,37 +143,225 @@ func (e *exporter) serializeMetrics(md pdata.Metrics) ([]string, int) {
+					for x := 0; x < metric.IntSum().DataPoints().Len(); x++ {
+						dp := metric.IntSum().DataPoints().At(x)
+						dims := labelsToDimensions(dp.LabelsMap())
+
+						valueOpt := dtMetric.WithIntCounterValueTotal(dp.Value())
+						if metric.IntSum().AggregationTemporality() == pdata.AggregationTemporalityDelta {
+							valueOpt = dtMetric.WithIntCounterValueDelta(dp.Value())
+						}
+
+						line, err := serializeMetric(
+							metric.Name(),
+							dtMetric.WithPrefix(e.cfg.Prefix),
+							dtMetric.WithDimensions(
+								dimensions.MergeLists(
+									e.defaultDimensions,
+									dimensions.NewNormalizedDimensionList(dims...),
+									e.staticDimensions,
+								),
+							),
+							dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+							valueOpt,
+						)
+
+						if err != nil {
+							e.logger.Sugar().Warnf("failed to serialize %s int sum %s - %s", metric.IntSum().AggregationTemporality().String(), metric.Name(), err.Error())
+							continue
+						}
+
+						metricLines = append(metricLines, line)
+					}
 				case pdata.MetricDataTypeDoubleSum:
-					l = serialization.SerializeDoubleDataPoints(name, metric.DoubleSum().DataPoints(), e.cfg.Tags)
+					for x := 0; x < metric.DoubleSum().DataPoints().Len(); x++ {
+						dp := metric.DoubleSum().DataPoints().At(x)
+						dims := labelsToDimensions(dp.LabelsMap())
+
+						valueOpt := dtMetric.WithFloatCounterValueTotal(dp.Value())
+						if metric.DoubleSum().AggregationTemporality() == pdata.AggregationTemporalityDelta {
+							valueOpt = dtMetric.WithFloatCounterValueDelta(dp.Value())
+						}
+
+						line, err := serializeMetric(
+							metric.Name(),
+							dtMetric.WithPrefix(e.cfg.Prefix),
+							dtMetric.WithDimensions(
+								dimensions.MergeLists(
+									e.defaultDimensions,
+									dimensions.NewNormalizedDimensionList(dims...),
+									e.staticDimensions,
+								),
+							),
+							dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+							valueOpt,
+						)
+
+						if err != nil {
+							e.logger.Sugar().Warnf("failed to serialize %s double sum %s - %s", metric.DoubleSum().AggregationTemporality().String(), metric.Name(), err.Error())
+							continue
+						}
+
+						metricLines = append(metricLines, line)
+					}
 				case pdata.MetricDataTypeIntHistogram:
-					l = serialization.SerializeIntHistogramMetrics(name, metric.IntHistogram().DataPoints(), e.cfg.Tags)
+					for x := 0; x < metric.IntHistogram().DataPoints().Len(); x++ {
+						if metric.IntHistogram().AggregationTemporality() == pdata.AggregationTemporalityDelta {
+							e.logger.Sugar().Warnf("failed to serialize delta int histogram %s - %s", metric.Name(), "unsupported temporality")
+						}
+
+						dp := metric.IntHistogram().DataPoints().At(x)
+						dims := labelsToDimensions(dp.LabelsMap())
+
+						// Because we do not know the actual min and max, we estimate them based on the min and max non-empty bucket
```

Comment:
What happens if all buckets are empty? Min == Max == lowest bucket bound?

Reply:
(see comment in tests as well) Are the bucket bounds as long as the bucket counts? Does the first bucket count describe everything below the first bucket bound, or above?